plugins/ui/nvim-highlight-colors.nix

```diff
@@ -0,0 +1,96 @@
+{
+  lib,
+  helpers,
+  config,
+  pkgs,
+  ...
+}:
+
+let
+  inherit (helpers.defaultNullOpts)
+    mkBool
+    mkEnum
+    mkListOf
+    mkStr
+    mkStr'
+    ;
+in
+helpers.neovim-plugin.mkNeovimPlugin config {
+  name = "nvim-highlight-colors";
+  defaultPackage = pkgs.vimPlugins.nvim-highlight-colors;
+
+  maintainers = [ helpers.maintainers.thubrecht ];
+
+  settingsOptions = {
```

Comment:
Note: because `settings` is a freeform option, there is no need to declare sub-options for _every_ upstream plugin option. Users can define any config they like, so having too many options can just end up being a maintenance burden.

The judgement call is yours to make though, I won't block a PR for having "too many" settings options :grin: